File: backend/app/schemas/cv.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class CVBase(BaseModel):
    title: str

class CVCreate(CVBase):
    pass

class CVResponse(CVBase):
    id: int
    user_id: int
    file_path: str
    file_name: str
    parsed_text: Optional[str] = None
    skills: Optional[str] = None
    experience: Optional[str] = None
    education: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    
    class Config:
        from_attributes = True

class CVAnalysis(BaseModel):
    skills: list
    experience: list
    education: list
    summary: str

File: frontend/src/types/CV.ts
export interface CV {
  id: number;
  user_id: number;
  title: string;
  file_path: string;
  file_name: string;
  parsed_text?: string;
  skills?: string;
  experience?: string;
  education?: string;
  created_at: string;
  updated_at?: string;
}

export interface CVCreate {
  title: string;
}

export interface CVAnalysis {
  skills: string[];
  experience: any[];
  education: any[];
  summary: string;
}
